File: fibonacci/csharp_fibonacci/fibonacci.cs
using System;

public class Fibonacci {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit(1);
    }

    for( int idx_i = 1; idx_i < 25; idx_i++ ) {
      Console.WriteLine("fibonacci(" + idx_i + ")=<" + fibonacci(idx_i) + ">");
    }
  }

  private static int fibonacci( int n ) {
    if( n == 0 ) {
      return 0;
    } else if( n == 1 ) {
      return 1;
    } else {
      return(fibonacci( n - 1 ) + fibonacci( n - 2 ));
    }
  }
}

File: fibonacci/fsharp_fibonacci/fibonacci.fs
#light
open System

//comment
(* commment *)

let rec fibonacci n = if n = 0 then 0 elif n = 1 then 1 else fibonacci(n - 1) + fibonacci(n - 2)

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    for idx_i in 1 .. 24 do
      printfn "fibonacci(%d)=<%d>" idx_i (fibonacci idx_i)
    0